test/parallel/test-inspector-exit-worker-in-wait-for-connection.mjs

```diff
@@ -0,0 +1,21 @@
+// Flags: --inspect=0
+import * as common from '../common/index.mjs';
+common.skipIfInspectorDisabled();
+
+import { isMainThread, Worker } from 'node:worker_threads';
+import { fileURLToPath } from 'node:url';
+import inspector from 'node:inspector';
+
+// Ref: https://github.com/nodejs/node/issues/52467
+// - worker.terminate() should terminate the worker and the pending
+//   inspector.waitForDebugger().
+if (isMainThread) {
+  const worker = new Worker(fileURLToPath(import.meta.url));
+  await new Promise((r) => setTimeout(r, 2_000));
```

Comment:
Using timeout to wait until a worker is ready can be flaky on slow CI machines.

Reply:
This could be more stable approach:

```diff
// Flags: --inspect=0
import * as common from '../common/index.mjs';
common.skipIfInspectorDisabled();

-import { isMainThread, Worker } from 'node:worker_threads';
+import { isMainThread, parentPort, Worker } from 'node:worker_threads';
import { fileURLToPath } from 'node:url';
import inspector from 'node:inspector';

// Ref: https://github.com/nodejs/node/issues/52467
// - worker.terminate() should terminate the worker and the pending
//   inspector.waitForDebugger().
if (isMainThread) {
  const worker = new Worker(fileURLToPath(import.meta.url));
- await new Promise((r) => setTimeout(r, 2_000));
+ await new Promise((r) => worker.on("message", r));

  worker.on('exit', common.mustCall());
  await worker.terminate();
} else {
  inspector.open();
+ parentPort.postMessage("inspector is open")
  inspector.waitForDebugger();
}
```

Reply:
Using MessagePort doesn't seem to be enough. I tried using `common.platformTimeout()` to adjust the timeout to give the worker more time.